cmake/options.cmake

```diff
@@ -22,7 +22,11 @@ option(WARNINGS_ARE_ERRORS  "Treat warnings as errors"
 option(EXTERNALS_AS_SYSTEM  "Treat externals as system includes"                            OFF)
 
 set(USE_MATCHCOMPILER "Auto" CACHE STRING "Usage of match compiler")
-set_property(CACHE USE_MATCHCOMPILER PROPERTY STRINGS Auto Off On Verify) 
+set(_MATCHCOMPILER_STRINGS Auto Off On Verify)
```

Comment:
According to readme.md this should be "ON" in capital letters. Same for "OFF".
Otherwise:
CMake Error at cmake/options.cmake:28 (message):
  Invalid USE_MATCHCOMPILER value 'ON'